File: forum-api-back/forum-api-back/DataAccess/DataObjects/Comment.cs
using System;
using System.Collections.Generic;

namespace forum_api_back.DataAccess.DataObjects
{
    public partial class Comment
    {
        public int Idcomment { get; set; }
        public string NomUtilisateur { get; set; } = null!;
        public string Contenu { get; set; } = null!;
        public DateTime DateCreation { get; set; }
        public DateTime? DateModification { get; set; }
        public int? TopicId { get; set; }

        public virtual Topic? Topic { get; set; }
    }
}

File: forum-front/src/app/models/comment.ts

export class Comment {

    idComment: Number;
    nomUtilisateur: string;
    contenu: string;
    dateCreation: Date;
    dateModification?:  Date;
    topicId: Number;

    constructor(idComment: Number, nomUtilisateur: string, contenu: string, dateCreation: Date, dateModification: Date, topicId: Number){
        this.idComment = idComment;
        this.nomUtilisateur = nomUtilisateur;
        this.contenu = contenu;
        this.dateCreation = dateCreation;
        this.dateModification = dateModification;
        this.topicId = topicId;
    }

}